crates/spirv-std/src/textures.rs

```diff
@@ -68,7 +68,7 @@ impl StorageImage2d {
     where
         I: Integer,
         V: Vector<I, N>,
-        V2: Vector<f32, N>,
+        V2: Vector<f32, 2>,
```

Comment:
Looks like you accidentally made the result be a 2-component vector instead of the coordinate.

Reply:
Well doesn't it say ["See the client API specification for handling of coordinates outside the image."](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#OpImageRead) for the coordinates?  So shouldn't coordinates be `N` to allow for that and the return vector should be `2` because it's a 2D unarrayed (is that how you say that?) image, so it returns Vector of 2?

Reply:
"coordinates outside the image" means if the coordinates are out-of-bounds, not if they're the wrong dimension. A 2d image will always take a Vec2 coordinate. Also, the size of the vector stored in the image has no correlation to the dimension of the image, e.g. a 2d image can store a 4-component pixel, it doesn't always have to store a Vec2.